### ISSUE TITLE
[Request] Allow developing locally

### ISSUE THREAD

**Issue by CaptainChemist** [abernix/serverless-graphql-lambda-simple#1]
This is an amazing repo, thanks so much for posting it. I have not seen any other serverless apollo-engine projects and I especially love how it uses aws fargate for its containers. I can really see this repo being the defacto boilerplate for graphQL backends.

To that end, would it be possible to add a npm task so that we can develop with it locally? My experience is primarily with graphQL-yoga package which [has a really nice lambda graphql function but it adds a http event to the graphQL handler in serverless.yml](https://github.com/CaptainChemist/serverlessDeals), which you aren't doing in this project and it isn't quite clear how I can make that conversion. 

Thanks!

**Comment by abernix** [comment]
Sorry I hadn't seen this earlier, this was a PoC that I never got to revisit!

That said, I think it should already be possible to test locally using (for example, for an `engineHealth` query):

    npx serverless invoke local -f query -d '{ "body": "{\"query\": \"{ engineHealth }\" }" }'

**Comment by CaptainChemist** [comment]
No problem, thanks so much for commenting! I'm not sure if this is something you ran into when trying to hook up a frontend apollo app to your server, but I found that I needed to handle HTTP requests other than POST, specifically OPTIONS but I think I may have even seen HEAD before. I suspect you might run into this too because I didn't see explicate handling of that type of request.

[This was my best attempt at handling it](https://github.com/CaptainChemist/serverlessDeals/blob/apollo-engine/handler.js)
it seems to be close except the annoying need to have to pass the callback into context and call that whenever I encounter an error since the lambdaGraphql package doesn't have a formatError that I can see. In any case, thanks again!
